h/services/list_groups.py

```diff
@@ -81,34 +119,31 @@ def _open_groups(self, user=None, authority=None):
                       .all())
         return self._sort(groups)
 
-    def _private_groups(self, user=None):
-        """Return this user's private groups per user.groups."""
+    def _user_groups(self, user=None):
+        """Return all groups that this user is a member of regardless of type"""
 
         if user is None:
             return []
         return self._sort(user.groups)
 
-    def _parse_origin(self, uri):
-        """
-        Return the origin of a URI or None if empty or invalid.
+    def _private_groups(self, user=None):
+        """Return all private groups that this user is a member of"""
 
-        Per https://tools.ietf.org/html/rfc6454#section-7 :
-        Return ``<scheme> + '://' + <host> + <port>``
-        for a URI.
+        user_groups = self._user_groups(user)
+        return [group for group in user_groups if group.type == 'private']
 
-        :param uri: URI string
+    def _scoped_groups(self, authority, document_uri):
```

Comment:
Note name change to reflect the fact that there isn't any meaningful difference between open scoped groups and restricted scoped groups in this context (a read-only context currently—both are world-readable).